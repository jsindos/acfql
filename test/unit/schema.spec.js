/* global expect, describe, it */

const ACFStore = require('../../generate/acfstore')
const { generateTemplate: generateSchemaTemplate } = require('../../src/schema/schema.template.js')
const { generateTemplate: generateCustomPostTypeTemplate } = require('../../src/schema/customPostTypes/CustomPostType.template.js')
const { generateTemplate: generateFieldGroupTemplate } = require('../../src/schema/fieldGroups/FieldGroup.template.js')

describe('generate schema files', () => {
  it('generateSchemaTemplate', () => {
    const a = new ACFStore()
    a.addCustomPostType('wp_apple')
    a.addFieldGroupRelation('wp_apple', 'Apple Information')
    a.addFieldGroup('Apple Information')
    expect(generateSchemaTemplate(a.customPostTypes, a.fieldGroups)).toMatchSnapshot()
  })

  it('generateCustomPostTypeTemplate', () => {
    const a = new ACFStore()
    a.addCustomPostType('wp_apple', 'post_type')
    a.addFieldGroupRelation('wp_apple', 'Apple Information', 'post_type')
    a.addFieldGroupRelation('wp_apple', 'Other Information', 'post_type')
    a.addFieldGroup('Apple Information')
    const templates = generateCustomPostTypeTemplate(a.customPostTypes, a.fieldGroups)
    expect(templates.find(t => t.fileName === 'Apple').template).toMatchSnapshot()
  })

  it('generateFieldGroupTemplate', () => {
    const a = new ACFStore()
    a.addCustomPostType('wp_apple')
    a.addFieldGroupRelation('wp_apple', 'Apple Information')
    a.addFieldGroupRelation('wp_apple', 'Other Information')
    a.addFieldGroup('Apple Information')
    a.addField('cultivar', 'Cultivar', 'Apple Information')
    a.addField('price', 'Price', 'Apple Information')
    const templates = generateFieldGroupTemplate(a.customPostTypes, a.fieldGroups)
    expect(templates.find(t => t.fileName === 'AppleInformation').template).toMatchSnapshot()
  })
})
