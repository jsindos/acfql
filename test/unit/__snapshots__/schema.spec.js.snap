// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate schema files generateCustomPostTypeTemplate 1`] = `
"const AppleInformation = require('../fieldGroups/AppleInformation')
const OtherInformation = require('../fieldGroups/OtherInformation')


const Apple = \`
type Apple {
  ID: Int
  postTitle: String
  
  appleInformation: AppleInformation,
  otherInformation: OtherInformation
}
\`

module.exports = () => [ Apple, AppleInformation, OtherInformation ]
"
`;

exports[`generate schema files generateFieldGroupTemplate 1`] = `
"const AppleInformation = \`

type AppleInformation {
  
  cultivar: JSON
  price: JSON
}
\`

module.exports = () => [ AppleInformation ]

"
`;

exports[`generate schema files generateSchemaTemplate 1`] = `
"
const Post = require('./Post')

const RootQuery = \`
  scalar JSON

  type Query {
    
    post(name: String, id: Int, postType: String): Post
    posts(category: String, postType: String, language: String): [Post]
  }
\`

const SchemaDefinition = \`
  schema {
    query: Query
  }
\`

module.exports = [
  
  Post,
  RootQuery,
  SchemaDefinition
]
"
`;
